src/perl/vdotest/VDOTest/Vdorecover.pm

```diff
@@ -228,18 +240,46 @@ sub testFullFilesystem {
   # Sleep to give the script time to start up.
   sleep(20);
 
-  # Remount the filesystem to clear it's errors.
-  $fs->unmount();
+  # We should now be sitting at the prompt from vdorecover (for lvmvdo device).
+  # Remount the filesystem, delete some data, and do a fstrim.
+  # Logged dmesg output should show start and completion of recovery.
   $fs->mount();
-
-  # Delete some data and do a fstrim.
   $dir2->rm(async => 0);
-  my $machine = $self->getDevice()->getMachine();
   $machine->assertExecuteCommand("sudo fstrim " . $fs->getMountDir());
   $self->syncDeviceIgnoringErrors();
+  $device->runOnHost("sudo dmesg -c", 1);
+  $fs->unmount();
+  $device->runOnHost("sudo dmesg -c >/dev/null");
 
   # Make sure we succeeded at recovering.
-  $recoverTask->result();
+
+  # It would be best to evaluate the success of the recovery script by verifying that
+  # 'Recovery process completed' was logged to stdout. However, this would require
+  # modifications to AsyncSub in order to make stdout accessible. Although it can be obtained
+  # from the RemoteMachine object by calling sendCommand() and getStdout() instead of
+  # assertExecuteCommand(), it is out of scope. 
+
+  # There is a built-in wait mechanism in AsyncSub::result() that waits for completion before
+  # returning the result. It does not appear to be sufficient, as we occasionally still fail to
+  # remount the filesystem.
+  my $i = 0;
+  my $recoveryCheck = sub {
+    $i++;
+    $log->debug("Recovery check iteration $i");
```

Comment:
Is there a benefit to printing the iteration number?  I don't feel strongly about removing this, but I have seen objections to counting things without much of a purpose behind it.

Reply:
Ultimately, no. But considering this test has a history of timing issues, I thought that it would be useful to have some insight into the number of retries we're going through in case it fails again.

Reply:
Once it consistently passes nightly testing, we could remove the iteration logging.

Reply:
Sure.  I doubt we'll do much revisiting voluntarily, though. :)

As I said, I don't feel strongly about removing it.  Just that we had just finished having a discussion about counting in a different context and seemed like it might end up being a sticking point.  In your testing, what's the highest iteration you've seen it reach at this point?

Reply:
Before I changed the retry command to wait 5 seconds in between attempts instead of 1, it was 15. Now it's 3.